test/libyul/yulOptimizerTests/unusedPruner/msize.yul

```diff
@@ -0,0 +1,9 @@
+{
+    let a := 1
+    let b := msize()
+    let c := mload(10)
+}
+// ====
+// step: unusedPruner
+// ----
+// { pop(mload(10)) }
```

Comment:
Couldn't `mload()` be safely be optimized out here because it is not being used? I was of the impression that not optimizing was reserved for programs like this

```
{
mstore(0, msize())
}
```

where the value returned by `msize()` is used by some other instruction that has a side-effect.

Reply:
You are right, a second run would probably remove it, so your test is better suited.